caikit_tgis_backend/tgis_backend.py

```diff
@@ -195,6 +197,50 @@ def get_connection(
                 self._model_connections.setdefault(model_id, model_conn)
         return model_conn
 
+    def register_model_connection(
+        self,
+        model_id: str,
+        conn_cfg: Optional[Dict[str, Any]] = None,
```

Comment:
We should have explicit logic to fill in missing fields in this inbound `conn_cfg` using the base `connection`. We could also have a bool arg that says whether or not to do that.

Reply:
Added `fill_with_defaults` arg.